### PR TITLE
Add type-safe EncodeObjects

### LINKED ISSUE #734: Make encode objects type-safe
E.g.
```ts
interface EncodeMsgDelegate extends EncodeObject {
  readonly typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
  readonly value: MsgDelegate,
}
```

See https://github.com/cosmos/cosmjs/pull/733#discussion_r601428521

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #734

### CODE REVIEW

File: packages/cosmwasm-stargate/src/queries/wasm.spec.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ import {
 } from "@cosmjs/stargate";
 import { assert, assertDefined } from "@cosmjs/utils";
 import Long from "long";
+import {
+  MsgExecuteContractEncodeObject,
+  MsgInstantiateContractEncodeObject,
+  MsgStoreCodeEncodeObject,
+} from "src/encodeobjects";
```
Comment (webmaster128) [inline packages/cosmwasm-stargate/src/queries/wasm.spec.ts:21]: Shouldn't this be a relative path?

File: packages/stargate/src/queries/distribution.spec.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import { coin, coins, DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
 import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
 import { sleep } from "@cosmjs/utils";
+import { MsgDelegateEncodeObject } from "src/encodeobjects";
```
Comment (webmaster128) [inline packages/stargate/src/queries/distribution.spec.ts:5]: relative path

File: packages/stargate/src/queries/staking.spec.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import { coin, coins, DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
 import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
 import { sleep } from "@cosmjs/utils";
+import { MsgDelegateEncodeObject, MsgUndelegateEncodeObject } from "src/encodeobjects";
```
Comment (webmaster128) [inline packages/stargate/src/queries/staking.spec.ts:5]: relative path

**Review by webmaster128 — APPROVED** [review]
Nice, just the imports

File: packages/cosmwasm-stargate/src/queries/wasm.spec.ts (hunk 1 above)
Comment (willclarktech) [reply packages/cosmwasm-stargate/src/queries/wasm.spec.ts:21]: Ugh, I have the feeling my VSCode used to use the import paths I wanted and recently switched to this.

**Pull request merged**